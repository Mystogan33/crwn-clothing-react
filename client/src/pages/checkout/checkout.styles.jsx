import styled from 'styled-components';

const CheckoutPageContainer = styled.div`
  width: 55%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 50px auto 0;

  button {
    margin-left: auto;
    margin-top: 50px;
  }
`;

/* padding: 10px 0; */
const CheckoutHeaderContainer = styled.div`
  width: 100%;
  height: 40px;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid darkgrey;
`;

const HeaderBlockContainer = styled.div`
  text-transform: capitalize;
  width: 23%;

  &:last-child {
    width: 8%;
  }
`;

const TotalContainer = styled.div`
  margin-top: 30px;
  margin-left: auto;
  font-size: 36px;
`;

const WarningContainer = styled.div`
  text-align: center;
  margin-top: 40px;
  font-size: 24px;
  color: red;
`;

export {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
  WarningContainer
};
