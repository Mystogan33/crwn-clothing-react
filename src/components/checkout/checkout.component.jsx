import React from 'react';
import './checkout.styles.scss';
