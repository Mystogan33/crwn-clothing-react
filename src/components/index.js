export * from './directory/directory.component';
export * from './menu-item/menu-item.component';
export * from './collection-preview/collection-preview.component';
export * from './collection-item/collection-item.component';
export * from './sign-in/sign-in.component';
export * from './custom-button/custom-button.component';
export * from './form-input/form-input.component';
export * from './header/header.component';
export * from './sign-up/sign-up.component';
export * from './cart-icon/cart-icon.component';
export * from './cart-dropdown/cart-dropdown.component';
export * from './cart-item/cart-item.component';
export * from './checkout-item/checkout-item.component';
export * from './collections-overview/collections-overview.component';
export * from './stripe-button/stripe-button.component';
export * from './with-spinner/with-spinner.component';
