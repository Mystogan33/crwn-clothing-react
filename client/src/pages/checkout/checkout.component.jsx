import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';

import { CheckoutItem, StripeCheckoutButton } from '../../components';

import {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
  WarningContainer
} from './checkout.styles';

const CheckoutPage = ({ cartItems, total }) => (
 <CheckoutPageContainer>
  <CheckoutHeaderContainer>
    <HeaderBlockContainer>
      <span>Product</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Description</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Quantity</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Price</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Remove</span>
    </HeaderBlockContainer>
  </CheckoutHeaderContainer>
  { cartItems.map(cartItem => (
      <CheckoutItem key={cartItem.id} cartItem={cartItem} />
    ))
  }
  <TotalContainer>TOTAL: ${total}</TotalContainer>
  <WarningContainer>
    *Please use the following test credit card for payments*
    <br />
    Card Number: 4242 4242 4242 4242 - CVC: 123 - Date: 10/2032
  </WarningContainer>
  <StripeCheckoutButton  price={total} />
 </CheckoutPageContainer>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal
});

export default connect(mapStateToProps)(CheckoutPage);
