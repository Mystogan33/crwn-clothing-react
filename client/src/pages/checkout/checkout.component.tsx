import React from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { RootState } from '../../redux/root-reducer';
import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';
import { ICartItems } from '../../interfaces/interfaces';

import { CheckoutItem, StripeCheckoutButton } from '../../components';

import {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
  WarningContainer
} from './checkout.styles';

interface selectors {
  cartItems: ICartItems,
  total: number
};

const mapStateToProps = createStructuredSelector<RootState, selectors>({
  cartItems: selectCartItems,
  total: selectCartTotal
});

const connector = connect(mapStateToProps);
type ReduxProps = ConnectedProps<typeof connector>;
type AppProps = ReduxProps;

export const CheckoutPage = ({ cartItems, total }: AppProps) => (
 <CheckoutPageContainer>
  <CheckoutHeaderContainer>
    <HeaderBlockContainer>
      <span>Product</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Description</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Quantity</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Price</span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>Remove</span>
    </HeaderBlockContainer>
  </CheckoutHeaderContainer>
  { cartItems.map(cartItem => (
      <CheckoutItem key={cartItem.id} cartItem={cartItem} />
    ))
  }
  <TotalContainer>TOTAL: ${total}</TotalContainer>
  <WarningContainer>
    *Please use the following test credit card for payments*
    <br />
    Card Number: 4242 4242 4242 4242 - CVC: 123 - Date: 10/2032
  </WarningContainer>
  <StripeCheckoutButton price={total} />
 </CheckoutPageContainer>
);

export default connector(CheckoutPage);
