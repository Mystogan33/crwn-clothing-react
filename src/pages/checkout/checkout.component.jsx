import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';

import { CheckoutItem } from '../../components';

import './checkout.styles.scss';

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal
});

export const CheckoutPage = connect(mapStateToProps)(
  ({ cartItems, total }) => (
     <div className="checkout-page">
      <div className="checkout-header">
        <div className="header-block">
          <span>Product</span>
        </div>
        <div className="header-block">
          <span>Description</span>
        </div>
        <div className="header-block">
          <span>Quantity</span>
        </div>
        <div className="header-block">
          <span>Price</span>
        </div>
        <div className="header-block">
          <span>Remove</span>
        </div>
      </div>
      { cartItems.map(cartItem => (
          <CheckoutItem key={cartItem.id} cartItem={cartItem} />
        ))
      }
      <div className="total">
        <span>TOTAL: ${total}</span>
      </div>
     </div>
  )
)
