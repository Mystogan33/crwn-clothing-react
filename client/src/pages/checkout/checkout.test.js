import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutPage } from './checkout.component';
import { CheckoutItem } from '../../components';

describe('checkout Component', () => {
  let wrapper;
  let mockCartItems = [];
  
  beforeEach(() => {
    const mockProps = {
      cartItems: mockCartItems,
      total: 100
    };

    wrapper = shallow(<CheckoutPage {...mockProps} />);
  });

  it('should render CheckoutPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the correct number of checkout item', () => {
    expect(wrapper.find(CheckoutItem).length).toEqual(mockCartItems.length);
  });
});