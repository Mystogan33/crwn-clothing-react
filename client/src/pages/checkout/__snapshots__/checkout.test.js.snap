// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout Component should render CheckoutPage component 1`] = `ShallowWrapper {}`;
